dygraph.go

```diff
@@ -0,0 +1,70 @@
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"html/template"
+)
+
+// Based on script from: https://github.com/tsenart/vegeta/blob/master/lib/plot/assets/plot.html.tpl#L24
+const chartScript = `document.getElementById("{{ .ID }}");
+  const o = {{ .Opts }}
+  const d = {{ .Data }}
+  const plot = new Dygraph(container, data, opts);`
+
+// GenerateChart creates a JS snippet that creates a Dygraph chart
+// based on given data and options. Dygraph charts require a container
+// to attach to, which can be identified by dom element id.
+//
+// TODO(abhi): Figure out reasonable defaults for dygraph options
+func GenerateChart(id string, data []byte, opts DygraphsOpts) (template.JS, error) {
+	t, err := template.New(fmt.Sprintf("%s dygraph chart", id)).Parse(chartScript)
+	if err != nil {
+		return "", err
+	}
+
+	optsJSON, err := json.MarshalIndent(&opts, "    ", " ")
```

Comment:
If controlling indentation is not a must, then consider passing the struct value directly to `Template.Execute`:

https://pkg.go.dev/html/template#hdr-A_fuller_picture

> Non-string values can be used in JavaScript contexts. If {{.}} is
> 
> ```
> struct{A,B string}{ "foo", "bar" }
> ```
> 
> in the escaped template
> 
> ```
> <script>var pair = {{.}};</script>
> ```
> 
> then the template output is
> 
> ```
> <script>var pair = {"A": "foo", "B": "bar"};</script>
> ```